lib/authlogic/session/cookies.rb

```diff
@@ -247,6 +273,36 @@ def generate_cookie_for_saving
           def destroy_cookie
             controller.cookies.delete cookie_key, domain: controller.cookie_domain
           end
+
+          def cookie_value
+            if protect_from_replay? && !remember_me?
+              @cookie_value ||= encrypt_cookie(raw_cookie)
+            else
+              @cookie_value ||= raw_cookie
+            end
+          end
+
+          def raw_cookie
+            remember_me_until_value = "::#{remember_me_until.iso8601}" if remember_me?
+            "#{record.persistence_token}::#{record.send(record.class.primary_key)}#{remember_me_until_value}"
+          end
+
+          def encrypt_cookie(value)
+            [aes_cipher(value, :encrypt)].pack('m')
+          end
+
+          def decrypt_cookie(value)
+            aes_cipher(value.unpack('m').first, :decrypt)
+          end
+
+          def aes_cipher(value, action = :encrypt)
+            action = :encrypt unless [:encrypt, :decrypt].include?(action)
+            aes = OpenSSL::Cipher::Cipher.new("AES-256-ECB")
+            aes.send(action)
+            aes.key = Digest::MD5.hexdigest(controller.session[:session_id])
```

Comment:
@sdsdkkk wouldn't this break because the `session[:session_id]` is lost when you close the browser, and the point of the Remember-Me cookie is to log you in when the browser is closed (or I might be misunderstanding this).

Reply:
Actually, what I wanted to achieve is to encrypt the user_credential token with a volatile value so it can't be saved and reused on a different browsing session.

If the remember me option is activated, the browser should get the raw value and the server should be able to recognize it so the remember me feature should work just fine, but the real user_credential would be exposed to the browser.